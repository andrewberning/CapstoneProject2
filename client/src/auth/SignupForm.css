.SignupForm {
  margin-top: 50px;
}

.SignupForm h2 {
  margin-bottom: 20px;
  text-align: center;
}

.SignupForm .btn {
  color: var(--light-clr);
  background-color: var(--secondary-accent-clr);
  border-color: black;
}

.SignupForm .btn:hover {
  background-color: var(--primary-accent-clr);
  border-color: black;
  color: var(--light-clr);
}