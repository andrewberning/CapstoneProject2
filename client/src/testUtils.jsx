import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
