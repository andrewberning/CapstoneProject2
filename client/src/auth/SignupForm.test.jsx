import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import SignupForm from "./SignupForm"
import { MemoryRouter } from 'react-router-dom';
import { UserProvider, mockSignup} from "../testUtils";


describe('SignupForm component', () => {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <UserProvider >
          <SignupForm />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider >
          <SignupForm />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls signup function with correct data when form is submitted', async () => {
    // Ensure mockSignup returns a successful signup response
    mockSignup.mockResolvedValue({ success: true});

    render(
      <MemoryRouter>
        <UserProvider >
          <SignupForm />
        </UserProvider>
      </MemoryRouter>
    );

    // Find the input elements
    const usernameInput = screen.getByLabelText(/username/i);
    const passwordInput = screen.getByLabelText(/password/i);
    const firstNameInput = screen.getByLabelText(/first name/i);
    const lastNameInput = screen.getByLabelText(/last name/i);
    const emailInput = screen.getByLabelText(/email/i);

    // Simulate user input
    fireEvent.change(usernameInput, { target: { value: 'testuser' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.change(firstNameInput, { target: { value: 'Test' } });
    fireEvent.change(lastNameInput, { target: { value: 'User' } });
    fireEvent.change(emailInput, { target: { value: 'testuser123@test.net' } });

    // Find and click the submit button
    const submitButton = screen.getByText(/submit/i);
    fireEvent.click(submitButton);

    // Assert that signup function is called with correct data
    await waitFor(() => {
      expect(mockSignup).toHaveBeenCalledWith({
        username: 'testuser',
        password: 'password123',
        firstName: 'Test',
        lastName: 'User',
        email: 'testuser123@test.net',
      });
    });
  });
});
