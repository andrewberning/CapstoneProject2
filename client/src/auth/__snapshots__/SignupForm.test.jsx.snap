// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignupForm component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        Sign Up
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                class="form-control"
                id="username"
                name="username"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="firstName"
              >
                First Name
              </label>
              <input
                class="form-control"
                id="firstName"
                name="firstName"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="form-control"
                id="lastName"
                name="lastName"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                type="email"
                value=""
              />
            </div>
            <button
              class="btn btn-primary float-right"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
